wiki/Client/File_formats/Osu_(file_format)/en.md

```diff
@@ -4,346 +4,623 @@
 
-#### Slider hitsounds
+#### Examples
 
-Apart from edge hitsounds, sliders also have an ongoing hitsound whenever the player is in range of the slider's follow circle. The sound file is looped for as long as it is active.
+`23,256,2608,2,0,L|34:151,1,105.000004005432,8|2,0:0|0:0,0:0:0:0:`
 
-This hitsound uses the hit object's `hitSound` and `hitSample` properties, but only the normal and whistle sounds are supported. Its filename is `<sampleSet>-slider<hitSound><index>.wav`, where `hitSound` is either `slide` for normal or `whistle` for whistle.
+`96,52,12600,6,1,B|200:200|250:200|250:200|300:150,2,310.123,2|1|2,0:0|0:0|0:2,0:0:0:0:`
 
-### Spinners
+#### Notice
 
-*Spinner syntax:* `x,y,time,type,hitSound,endTime,hitSample`
+`curve_type`
 
-- **`endTime` (Integer):** End time of the spinner, in milliseconds from the beginning of the beatmap's audio.
-- `x` and `y` do not affect spinners. They default to the centre of the playfield, `256,192`.
+> There are four types of slider curves in osu!:
+>
+> - `B` - Bézier - [Bézier curves](https://en.wikipedia.org/wiki/Bézier_curve) of arbitrary degree can be made. Multiple bézier curves can be joined into a single slider by repeating their points of intersection
+> - `C` - Centripetal catmull-rom - [Catmull curves](https://en.wikipedia.org/wiki/Centripetal_Catmull-Rom_spline) are an interpolating alternative to bézier curves. They are rarely used today due to their lack of visual appeal
+> - `L` - Linear - These curves form a straight path between all of their points
+> - `P` - Perfect circle - Perfect circle curves are limited to three points (including the hit object's position) that define the boundary of a circle. Using more than three points will result in the curve type being switched to bézier
 
-### Holds (osu!mania only)
+`curve_points`
 
-*Hold syntax:* `x,y,time,type,hitSound,endTime:hitSample`
+> In the second example, the Bézier curve slider is composite. The control points of the first curve are (96,52), (200,200) and (250,200), and the control points of the second curve are (250,200) and (300,150). Duplicated control points denote the [red anchor point](/wiki/Hit_object/Slider_anchor).
 
-- **`endTime` (Integer):** End time of the hold, in milliseconds from the beginning of the beatmap's audio.
-- `x` determines the index of the column that the hold will be in. It is computed by `floor(x * columnCount / 512)` and clamped between `0` and `columnCount - 1`.
-- `y` does not affect holds. It defaults to the centre of the playfield, `192`.
+`slides`
 
-### Examples
+> The slider is returnable if the value is more than `1`.
+
+`length`
+
+> If the slider's `length` is longer than the defined curve, the slider will extend until it reaches the target length:
+>
+> - For bézier, catmull, and linear curves, it continues in a straight line from the end of the curve
+> - For perfect circle curves, it continues the circular arc
```

Comment:
This is not true:
![screenshot227](https://user-images.githubusercontent.com/5307179/159925471-dc495b39-d4ff-42e3-a695-8c453fbbbfda.png)

Reply:
That was before my PR, I just formatted
![image](https://user-images.githubusercontent.com/35022026/159943047-58c534d0-d5c7-4b13-a548-a736df3db350.png)